crawler.py

```diff
@@ -124,6 +131,27 @@ def visit_title(*args, **kargs):
                         (self._fix_url(line.strip(), ""), 0))
         except IOError:
             pass
+    
+    # Returns the inverted index
+    def get_inverted_index(self):
+        return self.inverted_index
+
+    # Generates the resolved index everytime the function is called
+    def get_resolved_index(self):
+        resolvedIndex = dict()
+        for keys in self.inverted_index:
+            # Cycle through each word id of the inverted index, then find the corresponding word
+            # and use that as the key. Then, convert each of the set values of inverted_index to 
+            # their url form, and store that as value for the resolved index
+            resolvedIndex[self._id_to_word[keys]] = set([self._id_to_url[u] for u in self.inverted_index[keys]])
+        return resolvedIndex
+
+    # Adds a word_id to doc_id mapping. Creates a new set if the word hasn't been seen before
+    def update_inverted_index(self, word_id):
+        if word_id in self.inverted_index:
+            self.inverted_index[word_id].add(self._curr_doc_id)
+        else:
+            self.inverted_index[word_id] = set([self._curr_doc_id])
```

Comment:
A bit confused about the `self._curr_doc_id` part. Is this function called while crawling each document?

Should we explicitly pass in the doc id as a parameter? It might make testing the function easier